src/encoding.ts

```diff
@@ -0,0 +1,347 @@
+const BASE64_KEYSTR = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+
+function normalizeEncodingArgs(data: any): [ string, number[] ] {
```

Comment:
Since we're not using a custom ByteBuffer class, this method doesn't seem to be accomplishing much. We should just have the encoding functions only accept strings.